server/users.ts

```diff
@@ -878,23 +866,12 @@ export class User extends Chat.MessageContext {
 			// console.log('' + name + ' renaming: socket ' + i + ' of ' + this.connections.length);
 			connection.send(this.getUpdateuserText());
 		}
-		for (const roomid of this.games) {
-			const room = Rooms.get(roomid);
-			if (!room) {
-				Monitor.warn(`while renaming, room ${roomid} expired for user ${this.id} in rooms ${[...this.inRooms]} and games ${[...this.games]}`);
-				this.games.delete(roomid);
-				continue;
-			}
-			if (!room.game) {
-				Monitor.warn(`game desync for user ${this.id} in room ${room.roomid}`);
-				this.games.delete(roomid);
-				continue;
-			}
+		for (const room of Rooms.rooms.values()) {
+			if (!(oldid in room.users)) continue;
+			room.onRename(this, oldid, joining);
+			if (!room.game || !this.inGame(room)) continue;
 			room.game.onRename(this, oldid, joining, isForceRenamed);
```

Comment:
!

I wonder if our desyncs have been due to `onRename` not being called for players not in the room?

Reply:
I think it's probably worth calling it for either, here.

```suggestion
			if (oldid in room.users) {
				room.onRename(this, oldid, joining);
			}
			if (room.game && (oldid in room.users || oldid in room.game.playerTable)) {
				room.game.onRename(this, oldid, joining, isForceRenamed);
			}
```